src/testing/tree_construction/fixture.rs

```diff
@@ -15,37 +15,35 @@ pub struct FixtureFile {
 }
 
 /// Reads a given test file and extract all test data
-pub fn read_fixture_from_path(path: &PathBuf) -> Result<FixtureFile, Error> {
-    let input = fs::read_to_string(path)?;
-    let path = path.to_string_lossy().into_owned();
+pub fn read_fixture_from_path(path: impl AsRef<Path>) -> Result<FixtureFile, Error> {
```

Comment:
Not that I'm disagreeing, but for my own knowledge (still new to rust) - what's the benefit of using `AsRef` here versus a "regular" `&` reference?

Reply:
Read through the rust docs, I guess it's for anything that can be converted to a `PathBuf` reference. I'm not familiar enough with this portion of the code to know if that's done, but I don't see the harm

Reply:
> Read through the rust docs, I guess it's for anything that can be converted to a PathBuf reference. I'm not familiar enough with this portion of the code to know if that's done, but I don't see the harm

Exactly, I guess it isn't necessary at this point, I'm also not familiar with the code, so I only "_fixed_" things in a way that is common in Rust and I would also do myself. In this case it makes the api more flexible for future usage.